swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -446,32 +403,61 @@ extension Adapter: CallbackHandlerDelegate {
     }
   }
 
-  private func getSystemDefaultResolvers(interfaceName: String?) -> [String] {
+  private func setSystemDefaultResolvers(_ path: Network.NWPath) {
+    // Step 1: Get system default resolvers
     #if os(macOS)
       let resolvers = self.systemConfigurationResolvers.getDefaultDNSServers(
-        interfaceName: interfaceName)
+        interfaceName: path.availableInterfaces.first?.name)
     #elseif os(iOS)
       let resolvers = resetToSystemDNSGettingBindResolvers()
     #endif
 
+    // Step 2: Validate and strip scope suffixes
     var parsedResolvers: [String] = []
 
-    // Normalize addresses to remove any possible scope suffixes
     for stringAddress in resolvers {
       if let ipv4Address = IPv4Address(stringAddress) {
-        parsedResolvers.append("\(ipv4Address)")
+        if ipv4Address.isWithinSentinelRange() {
```

Comment:
Connlib filters these out for you already so you wouldn't need to do that :)

Reply:
It would be helpful to know that we are (mistakenly) trying to set the sentinels as resolvers. Note this is really only a problem on iOS AFAIK.

Reply:
I don't want to add the warning log to connlib because we may have intermitten states where we don't have a resolver so there would be false-positives.

All of this is going to get a lot easier once we do #8263 because then there will only be one resolver IP and that is simply always set. Then, connlib definitely knows if it is in a state where it doesn't have any servers to forward the queries to.